Volar support

<!-- Your feature may already be reported!
Please search on the issue tracker before creating one. -->

#### Describe your feature request
Please support Volar for Vue 3 as Vetur only supports Vue 2.

Volar was mentioned in [this pull request](https://github.com/helix-editor/helix/pull/2043) but I wonder why it could not work.
Things I have tried:
1. Installing the language server through `npm install -g @volar/vue-language-server typescript`
2. Adding this into `.config/helix/languages.toml`
```toml
[[language]]
name = "vue"
file-types = ["vue"]
language-server = { command = "vue-language-server", args = ["--stdio"] }
```
3. Verifying the installation with `hx --health`

However, opening a `.vue` file throws this into the logs
```
2022-04-20T21:12:09.123 helix_lsp::transport [INFO] -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"snippetSupport":false},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":false}},"window":{"workDoneProgress":true},"workspace":{"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false}}},"processId":4503,"rootUri":"file:///home/cheeze/repos/test"},"id":0}
2022-04-20T21:12:09.454 helix_lsp::transport [INFO] <- {"jsonrpc":"2.0","id":0,"error":{"code":-32603,"message":"Request initialize failed with message: Cannot read properties of undefined (reading 'documentFeatures')"}}
2022-04-20T21:12:09.454 helix_lsp::transport [ERROR] <- Internal error: Request initialize failed with message: Cannot read properties of undefined (reading 'documentFeatures')
2022-04-20T21:12:09.454 helix_lsp [ERROR] failed to initialize language server: protocol error: Internal error: Request initialize failed with message: Cannot read properties of undefined (reading 'documentFeatures')
2022-04-20T21:12:14.602 helix_lsp::transport [INFO] Language server not initialized, delaying request
```
I feel like it has something to do with the default value of `init_options` as stated [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#volar).
I am using Helix [release 22.03](https://github.com/helix-editor/helix/releases/tag/22.03)

I'm having similar issues, my config so far is
```bash
❯ cat ~/.config/helix/languages.toml
───────┬────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: /home/niedzwiedz/.config/helix/languages.toml
───────┼────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ [[language]]
   2   │ name = "vue"
   3   │ file-types = ["vue"]
   4   │ language-server = { command = "vue-language-server", args = ["--stdio"] }
───────┴─────────────────────────────────────────────────────────────────────────────────
```

and I'm getting 
```
2022-06-18T19:37:10.129 helix_lsp [ERROR] failed to initialize language server: protocol error: Internal error: Request initialize failed with message: Cannot read property 'typescript' of undefined
```

```toml
[[language]]
name = "vue"
file-types = ["vue"]
language-server = { command = "vue-language-server", args = ["--stdio"] }
scope = "text.html.vue"
roots = ["package.json"]
injection-regex = "vue"
config = {typescript = {serverPath = "/home/niedzwiedz/Pieprzyk/prometeusz/prometeusz-front/node_modules/typescript/lib/tsserverlibrary.js"}}}

```

this works but 
1. typescript server path needs to be hardcoded, in my case it's the one that is installed in my project
2. running ANY code action results in an error
```
2022-06-18T22:03:26.181 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/codeAction
2022-06-18T22:03:29.438 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/codeAction
2022-06-18T22:03:34.971 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/definition
2022-06-18T22:03:37.234 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/definition
2022-06-18T22:03:39.781 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/definition
2022-06-18T22:04:10.482 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/definition
2022-06-18T22:04:11.506 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/definition
2022-06-18T22:04:13.788 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/codeAction
2022-06-18T22:04:18.995 helix_lsp::transport [ERROR] <- MethodNotFound: Unhandled method textDocument/codeAction

```

from what I remember vetur had an issue where it would not support `textDocument/*` methods at all, only `workspace/*` or something like this... gonna investigate some more

--- a/book/src/generated/lang-support.md
+++ b/book/src/generated/lang-support.md
@@ -159,7 +159,7 @@
 | verilog | ✓ | ✓ |  | `svlangserver` |
 | vhdl | ✓ |  |  | `vhdl_ls` |
 | vhs | ✓ |  |  |  |
-| vue | ✓ |  |  | `vls` |
+| vue | ✓ |  |  | `vue-language-server` |
 | wast | ✓ |  |  |  |
 | wat | ✓ |  |  |  |
 | wgsl | ✓ |  |  | `wgsl_analyzer` |

--- a/languages.toml
+++ b/languages.toml
@@ -73,7 +73,7 @@ vlang-language-server = { command = "v", args = ["ls"] }
 vscode-css-language-server = { command = "vscode-css-language-server", args = ["--stdio"], config = { "provideFormatter" = true }}
 vscode-html-language-server = { command = "vscode-html-language-server", args = ["--stdio"], config = { provideFormatter = true } }
 vscode-json-language-server =  { command = "vscode-json-language-server", args = ["--stdio"], config = { provideFormatter = true } }
-vuels = { command = "vls" }
+vuels = { command = "vue-language-server", args = ["--stdio"], config = { typescript = { tsdk = "node_modules/typescript/lib/" } } }
 wgsl_analyzer = { command = "wgsl_analyzer" }
 yaml-language-server = { command = "yaml-language-server", args = ["--stdio"] }
 zls = { command = "zls" }
@@ -921,7 +921,7 @@ name = "vue"
 scope = "source.vue"
 injection-regex = "vue"
 file-types = ["vue"]
-roots = ["package.json", "vue.config.js"]
+roots = ["package.json"]
 indent = { tab-width = 2, unit = "  " }
 language-servers = [ "vuels" ]
 

